### PR TITLE
Feature: Improve reports ui

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
After playing with the improved reports UI from #7000, I still wasn't happy with it. I've since redesigned the form for adding report notes, and added the "ActionLog" history (in a rough form) so we can see all the actions taken in relation to a specific report.

Note: I wasn't sure on the right way to implement fetching the ActionLog, so went with a relation, but now I think a Finder method would be better, as it needs to pull from the Report, the statuses of the report and the reported user; Though I'm not sure on the best way to implement this.

Here's what the page now looks like:

localhost_3000_admin_reports_2 2

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
I really want to try to get the additional ActionLog items that are relevant into the report, but I'm not sure how to do this. Open to suggestions.

**Comment by nightpool** [comment]
it would be really good if the moderation log and action log displayed
interleaved with each other I think

**Comment by ThisIsMissEm** [comment]
@nightpool oh, I agree, but I've absolutely no idea how to do that in rails. Not to mention that it's not the full action log for now, which I'm struggling to build.

**Comment by Gargron** [comment]
I'm liking this a lot. :+1:

File: app/models/report.rb [hunk 1]
```diff
@@ -39,4 +39,36 @@ def statuses
...
+    report_log = Admin::ActionLog.where(target_type: 'Report', target_id: id).unscope(:order)
+    target_account_log = Admin::ActionLog.where(target_type: 'Account', target_id: target_account_id).unscope(:order)
+    statuses_log = Admin::ActionLog.where(target_type: 'Status', target_id: status_ids).unscope(:order)
+
+    sql = Admin::ActionLog.connection.unprepared_statement {
+      "((#{report_log.to_sql}) UNION ALL (#{target_account_log.to_sql}) UNION ALL (#{statuses_log.to_sql})) as admin_action_logs"
+    }
+
+    Admin::ActionLog.from(sql)
+  end
```
Comment (ThisIsMissEm) [inline app/models/report.rb]: This probably isn't ideal or perfect, so I'm open to suggestions here @nightpool @Gargron

File: app/models/report.rb [hunk 2]
```diff
@@ -39,4 +39,49 @@ def statuses
...
+    ).unscope(:order)
+
+    statuses_log = Admin::ActionLog.where(
+      target_type: 'Status',
+      target_id: status_ids,
+      created_at: created_at..updated_at
+    ).unscope(:order)
+
+    query = "((#{report_log.to_sql}) UNION ALL (#{target_account_log.to_sql}) UNION ALL (#{statuses_log.to_sql})) as admin_action_logs"
+    sql = Admin::ActionLog.connection.unprepared_statement(query)
```
Comment (Gargron) [inline app/models/report.rb]: It seems that this method takes a block argument, i.e. `unprepared_statement { query }`, but I am not totally sure why unprepared_statement is needed in here?

**Comment by ThisIsMissEm** [comment]
@Gargron this is just what I found when looking up how to do union queries with rails; there's probably a better way to do this, though I'm a bit out of my depth with rails knowledge needed.

File: app/models/report.rb (hunk 2 above)
Comment (Gargron) [reply app/models/report.rb]: We may want to just use https://github.com/brianhempel/active_record_union to do this the clean way

**Comment by Gargron** [comment]
Here is a fix for the UNION stuff that works for me:

```diff
diff --git a/app/models/report.rb b/app/models/report.rb
index e1bfa99..5b90c7b 100644
--- a/app/models/report.rb
+++ b/app/models/report.rb
@@ -61,27 +61,28 @@ class Report < ApplicationRecord
   end
 
   def history
-    report_log = Admin::ActionLog.where(
-      target_type: 'Report',
-      target_id: id,
-      created_at: created_at..updated_at
-    ).unscope(:order)
-
-    target_account_log = Admin::ActionLog.where(
-      target_type: 'Account',
-      target_id: target_account_id,
-      created_at: created_at..updated_at
-    ).unscope(:order)
-
-    statuses_log = Admin::ActionLog.where(
-      target_type: 'Status',
-      target_id: status_ids,
-      created_at: created_at..updated_at
-    ).unscope(:order)
-
-    query = "((#{report_log.to_sql}) UNION ALL (#{target_account_log.to_sql}) UNION ALL (#{statuses_log.to_sql})) as admin_action_logs"
-    sql = Admin::ActionLog.connection.unprepared_statement(query)
-
-    Admin::ActionLog.from(sql)
+    time_range = created_at..updated_at
+
+    sql = [
+      Admin::ActionLog.where(
+        target_type: 'Report',
+        target_id: id,
+        created_at: time_range
+      ).unscope(:order),
+
+      Admin::ActionLog.where(
+        target_type: 'Account',
+        target_id: target_account_id,
+        created_at: time_range
+      ).unscope(:order),
+
+      Admin::ActionLog.where(
+        target_type: 'Status',
+        target_id: status_ids,
+        created_at: time_range
+      ).unscope(:order),
+    ].map { |query| "(#{query.to_sql})" }.join(' UNION ALL ')
+
+    Admin::ActionLog.from("(#{sql}) AS admin_action_logs")
   end
 end
```

**Comment by ThisIsMissEm** [comment]
@Gargron this is now ready for review again, the 1 issue to fix from code climate isn't an issue, because we know what we're doing™

**Review by Gargron — APPROVED** [review]

**Pull request merged**